packages/rum-react/src/get-apm-routes.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * MIT License
+ *
+ * Copyright (c) 2017-present, Elasticsearch BV
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+import React from 'react'
+import { useFinishTransactionEffect } from './finish-transaction-effect'
+import {
+  Routes,
+  useLocation,
+  useNavigationType,
+  matchRoutes,
+  createRoutesFromChildren
+} from 'react-router-dom'
+import hoistNonReactStatics from 'hoist-non-react-statics'
+
+function getApmRoutes(apm) {
+  function ApmRoutes(props) {
+    const location = useLocation()
+    const navigationType = useNavigationType()
+
+    React.useLayoutEffect(
+      () => {
+        const routes = createRoutesFromChildren(props.children)
+        handleRouteChange(apm, routes, location)
+      },
+      // query param changes should not trigger this effect
+      // that's why we use `location.pathname` instead of `location`
+      [location.pathname, navigationType, props.children]
```

Comment:
query params might not affect it, can you add a test where we add a new children based on query param?

Reply:
I added that change because that was something we had in react-router 5

The code was this:

https://github.com/elastic/apm-agent-rum-js/blob/aeb75dd282a999b684f530fd0c3955bad40e6deb/packages/rum-react/src/get-apm-route.js#L60

And this was the test:

https://github.com/elastic/apm-agent-rum-js/blob/aeb75dd282a999b684f530fd0c3955bad40e6deb/packages/rum-react/test/specs/get-apm-route.spec.js#L186


The test for react-router 6 is this one:

https://github.com/elastic/apm-agent-rum-js/blob/f63ebb8a86214d694e3915d687e0c5290a746ffa/packages/rum-react/test/specs/get-apm-routes.spec.js#L137

Reply:
Sorry if i was not clear, The `location.pathname` is correct. I was asking about the `props.children`. There are cases when component adds say `<div>Current Query: ${path.query}>` that would trigger a transaction. If yes, lets add a test to cover this case. Because our current test uses query param but does not update the component children itself.